moped-editor/src/auth/user.js

```diff
@@ -149,20 +169,27 @@ export const useUser = () => {
   return context;
 };
 
-export const getJwt = user => {
-  return user.signInUserSession.idToken.jwtToken;
-};
+export const getJwt = user => user.idToken.jwtToken;
 
 export const isUserSSO = user =>
-  user.signInUserSession.idToken.payload["cognito:username"].startsWith(
-    "azuread_"
-  );
+  user.idToken.payload["cognito:username"].startsWith("azuread_");
+
+export const availableSessionTime = user => {
+  try {
+    return (
+      user.idToken.getExpiration() - Math.round(new Date().getTime() / 1000)
+    );
+  } catch {
+    return 0;
+  }
+};
+
+export const isSessionExpired = user => availableSessionTime(user) > 0;
```

Comment:
This code looks really similar to the logic you updated in `_isAuthenticated` and I don't see this new `isSessionExpired` boolean being used anywhere. Totally cool if we just leave `availableSessionTime` and `isSessionExpired` in here in this User Hook for future use but also don't want the logic in this implementation to stray from the primary logic being used in `_isAuthenticated` so maybe its better to delete the unused code to prevent future code cruft? I may be misinterpreting the code and up to you on if you want to address this comment.

Reply:
Thanks @mateoclarke I had a little debate in my mind and I left it there in case we need to revisit in the future, but now that you mention going stray from `_isAuthenticated` I've changed my mind. I am going to remove this helper method, and fix the new conflicts.